src/api/app/models/download_repository.rb

```diff
@@ -15,7 +15,7 @@ class DownloadRepository < ApplicationRecord
   def architecture_inclusion
     # Workaround for rspec validation test (validate_presence_of(:repository_id))
     return unless repository
-    return if repository.architectures.pluck(:name).include?(arch)
+    return if repository.repository_architectures.any? { |a| a.architecture.name == arch }
```

Comment:
that shouts N+1 to me.

How about repository.repository_architectures.where(architecture: Architecture.where(name: arch)).exists?

that translates into e.g.
```
  RepositoryArchitecture Exists (0.8ms)  SELECT  1 AS one FROM `repository_architectures` WHERE `repository_architectures`.`repository_id` = 1853016 AND `repository_architectures`.`architecture_id` IN (SELECT `architectures`.`id` FROM `architectures` WHERE `architectures`.`name` = 'x86_64') LIMIT 1
```

while your code translates into:

```
  RepositoryArchitecture Load (0.5ms)  SELECT `repository_architectures`.* FROM `repository_architectures` WHERE `repository_architectures`.`repository_id` = 98 ORDER BY position
  Architecture Load (0.4ms)  SELECT  `architectures`.* FROM `architectures` WHERE `architectures`.`id` = 2 LIMIT 1
  Architecture Load (0.4ms)  SELECT  `architectures`.* FROM `architectures` WHERE `architectures`.`id` = 1 LIMIT 1
```

(and for completness the original code):
```
   (0.9ms)  SELECT `architectures`.`name` FROM `architectures` INNER JOIN `repository_architectures` ON `architectures`.`id` = `repository_architectures`.`architecture_id` WHERE `repository_architectures`.`repository_id` = 98 ORDER BY position

Reply:
` repository.repository_architectures.where(architecture: Architecture.where(name: arch)).exists? ` wouldn't work with deep_cloneable, because the related architectures are not necessarily saved yet (when cloning they are not).

Reply:
then better stay with pluck but pluck the architecture_id

Reply:
This code runs at a time when the data might not be saved yet. So we can not use `pluck` to fetch thing from the DB.

Reply:
That worked... though I used the architecture_id instead so that we don't fetch the architecture each time.

Reply:
ah, I only checked the 'in memory' case.